test/unit/lib/plugins/aws/invoke.test.js

```diff
@@ -332,62 +199,105 @@ describe('test/unit/lib/plugins/aws/invoke.test.js', () => {
         raw: true,
       },
       awsRequestStubMap: {
-        // Stub AWS SDK invocation, and confirm `Payload` param
+        Lambda: {
+          invoke: (args) => {
+            lambdaInvokeStub.returns('payload');
+            return lambdaInvokeStub(args);
+          },
+        },
       },
     });
-    // Replaces
-    // https://github.com/serverless/serverless/blob/537fcac7597f0c6efbae7a5fc984270a78a2a53a/test/unit/lib/plugins/aws/invoke.test.js#L131-L138
+    expect(lambdaInvokeStub.args[0][0]).to.deep.equal({
+      FunctionName: result.serverless.service.getFunction('callback').name,
+      InvocationType: 'RequestResponse',
+      LogType: 'None',
+      Payload: Buffer.from(JSON.stringify('{"inputKey":"inputValue"}')),
+    });
   });
 
-  xit('TODO: should support JSON file path as data', async () => {
-    await runServerless({
+  it('should support JSON file path as data', async () => {
+    const lambdaInvokeStub = sinon.stub();
+    const result = await runServerless({
       fixture: 'invocation',
       command: 'invoke',
       options: {
         function: 'callback',
         path: 'payload.json',
       },
       awsRequestStubMap: {
-        // Stub AWS SDK invocation, and confirm `Payload` param
+        Lambda: {
+          invoke: (args) => {
+            lambdaInvokeStub.returns('payload');
+            return lambdaInvokeStub(args);
+          },
+        },
       },
     });
-    // Replaces
-    // https://github.com/serverless/serverless/blob/537fcac7597f0c6efbae7a5fc984270a78a2a53a/test/unit/lib/plugins/aws/invoke.test.js#L140-L154
+    expect(lambdaInvokeStub.args[0][0]).to.deep.equal({
+      FunctionName: result.serverless.service.getFunction('callback').name,
+      InvocationType: 'RequestResponse',
+      LogType: 'None',
+      Payload: Buffer.from(JSON.stringify({ dataInputKey: 'dataInputValue' })),
+    });
   });
 
-  xit('TODO: should support absolute file path as data', async () => {
-    await runServerless({
+  it('should support absolute file path as data', async () => {
+    const lambdaInvokeStub = sinon.stub();
+    const utils = new Utils();
+    const dataFilePath = path.join(getTmpDirPath(), 'data.json');
+    const data = { testProp: 'testValue' };
+    utils.writeFile(dataFilePath, JSON.stringify(data));
+    const result = await runServerless({
       fixture: 'invocation',
       command: 'invoke',
       options: {
         function: 'callback',
-        path: '' /* TODO: Pass absolute path to payload.json in fixture */,
```

Comment:
`TODO: Pass absolute path to payload.json in fixture` - I didn't quite fully understand what this meant (add the path to the file _within_ the json file?). 

I know writing files may not be ideal, so please let me know if there's a better approach

Reply:
We could avoid writing the file and instead reuse the one that is already present in the `invocation` fixture. In order to construct path to it, we would have to change the approach a little bit here and instead of running `serverless` directly, we will first setup the fixture and then execute `runServerless`.  It would be something along those lines:

```
const { servicePath: serviceDir } = await fixtures.setup('invocation');
const pathToPayload = path.join(serviceDir, 'payload.json');
await runServerless({
  cwd: serviceDir,
  <all other properties with the exeception of `fixture`>
});
```

You can see an example of such setup e.g. here: https://github.com/serverless/serverless/blob/b86128b2503e3df18c260264721bd913e676ada4/test/unit/lib/plugins/package/lib/packageService.test.js#L298

Reply:
perfect, thanks for that! I've updated to follow this approach 💯